packages/reactive-element/src/decorators/query-assigned-nodes.ts

```diff
@@ -38,21 +82,46 @@ import {decorateProperty} from './base.js';
  *   }
  * }
  * ```
+ *
+ * Note the type of this property should be annotated as `Array<Node>` if used
+ * without a `selector` or `Array<HTMLElement>` if a selector is provided.
+ *
+ * @param slotName A string name of the slot.
+ * @param flatten A boolean which when true flattens the assigned nodes,
+ *     meaning any assigned nodes that are slot elements are replaced with their
+ *     assigned nodes.
+ * @param selector A CSS selector used to filter the elements returned.
+ *
  * @category Decorator
+ * @deprecated Prefer passing in a single options object, i.e. `{slot: 'list'}`.
  */
 export function queryAssignedNodes(
-  slotName = '',
-  flatten = false,
-  selector = ''
+  slotName?: string,
+  flatten?: boolean,
+  selector?: string
+): TSDecoratorReturnType;
+
+export function queryAssignedNodes(
+  arg0?: string | QueryAssignedNodesOptions,
+  arg1?: boolean,
+  arg2?: string
 ) {
+  const {slot = '', selector = ''} =
+    typeof arg0 === 'object' ? arg0 : {slot: arg0 ?? '', selector: arg2 ?? ''};
```

Comment:
I think this might be a bit more readable without the destructuring. Doesn't create garbage objects either.

```ts
let slot = slotOrOptions;
let assignedNodesOptions = slotOrOptions;
if (typeof slotOrOptions === 'object') {
  slot = slotOrOptions.slot;
  selector = slotOrOptions.selector;
} else {
  assignedNodesOptions = {flatten};
}
```

Reply:
Done - although this option does require a type hint for `selector` in the implementation. Also prefer not creating unnecessary garbage.